superset/assets/src/dashboard/reducers/getInitialState.js

```diff
@@ -43,15 +43,56 @@ export default function(bootstrapData) {
 
   const dashboard = { ...bootstrapData.dashboard_data };
   let filters = {};
-  try {
-    // allow request parameter overwrite dashboard metadata
-    filters = JSON.parse(
-      getParam('preselect_filters') || dashboard.metadata.default_filters,
-    );
-  } catch (e) {
-    //
+
+  const getFiltersFromFilterConfig = (filter_configs) => {
+    const filters = {};
+    filter_configs.forEach( filter => {
+      const defaultValue = filter.hasOwnProperty("defaultValue")?filter["defaultValue"]:"";
+      var valArr = defaultValue!==""?defaultValue.split(";"):[]; 
+      if(valArr.length > 0){
+        filters[filter["column"]] = valArr;
+      }
+    })
+    return filters;
   }
 
+  // update filter with filter_box Viz default values
+  dashboard.slices.forEach(slice => {
+     let defultFilters = {};
+
+      // As per the current support only filter_box can publish global default filters
+      if(slice.form_data.viz_type == "filter_box"){
+
```

Comment:
can we pick filter_box value from const files

Reply:
We don't have any constant file regarding all VIZ types

Reply:
k